src/types.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+export type ErrorBoundaryState = {
+  error: Error | null;
+  componentStack: string | null;
+};
+
+// follows react-error-boundary https://github.com/bvaughn/react-error-boundary
+export type FallbackProps = ErrorBoundaryState & {
+  // this is our own addition
+  tryRecovering(): void;
+};
+
+export type ErrorBoundaryProps<P extends object = {}> = {
+  onError?: ErrorEventHandler;
+  FallbackComponent?: React.ComponentType<FallbackProps>;
+
+  // these are our own additions
+  recoverableProps?: Array<keyof P>;
+  reportErrors?: boolean; // true by default
+};
+
+export type ErrorEventHandler = (error: Error, componentStack: string) => void;
+
+// todo: this should be defined in a different repo concerning reporting
+export type Reporter = any; // any for now
```

Comment:
@philsturgeon do we have any concrete plans on having a shared for reporting such as `@stoplight/reporter` or would `@stoplight/types` be a good spot for now?

Reply:
I think a shared JS library which is a thin facade, which accepts some sort of JS logger so it can auto-log all errors reported, and some config items like ENV for SENTRY_TOKEN, etc. This should be shared so app/api can also use it.